### PR TITLE
chore: Enable multi-os golanglint-ci

### PR DESCRIPTION
**Pull request by powersj**
fixes: #13036

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -66,6 +66,22 @@ commands:
                 key: windows-go-<< parameters.cache_version >>-{{ checksum "go.sum" }}
             - run: 'sh ./scripts/installgo_windows.sh'
             - run: choco install mingw
+      - run: go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.53.2
```
Comment (zak-pawel) [inline .circleci/config.yml:69]: Does this approach use cache as `golangci-lint-action@v3`? 
I mean cache for `golangci-lint` not for `go` itself.

File: .circleci/config.yml [hunk 2]
```diff
@@ -66,6 +66,22 @@ commands:
...
+      - when:
+          condition:
+            equal: [ darwin, << parameters.os >> ]
+          steps:
+            - run: $HOME/go/bin/golangci-lint run
+      - when:
+          condition:
+            equal: [ windows, << parameters.os >> ]
+          steps:
+            - run: $HOME/go/bin/golangci-lint.exe run
```
Comment (zak-pawel) [inline .circleci/config.yml:84]: `--timeout 15m0s --verbose` for every run?

**Review by zak-pawel — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (powersj) [reply .circleci/config.yml:69]: The reason I was exploring using circle ci is this action was taking nearly 20mins on windows and macos during the first run. The second run that I remember was 3mins on linux and I think 5-6mins. on macos/windows. Compared to circleci, where these actions were adding 5mins in these same OSes. CircleCI ran into other issues on linux with the garbage collector.

Please keep in mind that neither circleci or macos are free either ;) I am going to update this PR to go back tot he GitHub actions for now and we will see how our bill looks after talking to internal folks.

**Review by srebhan — APPROVED** [review]
LGTM. Thanks @powersj!

**Pull request merged**